go/vt/vtgate/semantics/analyzer.go

```diff
@@ -53,7 +53,7 @@ type analyzer struct {
 // newAnalyzer create the semantic analyzer
 func newAnalyzer(dbName string, si SchemaInformation, fullAnalysis bool) *analyzer {
 	// TODO  dependencies between these components are a little tangled. We should try to clean up
-	s := newScoper()
+	s := newScoper(si.GetAggregateUDFs())
```

Comment:
any reason to keep udfs like this in the scoper, can they not be checked when needed?

Reply:
either we keep a slice of the udf names in the scoper, or we store the SchemaInfo and query later. No big difference imho, but I'm happy to change if you think the other way is better